packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -1750,4 +1751,28 @@ export class ProjectService {
         });
         return charts;
     }
+
+    async getSpaces(
+        user: SessionUser,
+        projectUuid: string,
+    ): Promise<SpaceSummary[]> {
+        const { organizationUuid } = await this.projectModel.get(projectUuid);
+        if (
+            user.ability.cannot(
+                'view',
+                subject('Project', { organizationUuid, projectUuid }),
+            )
+        ) {
+            throw new ForbiddenError();
+        }
+
+        const spaces = await this.spaceModel.find({ projectUuid });
+        const allowedSpaces = spaces.filter(
```

Comment:
Should we also check CASL `view space` ability ?

Reply:
This is the logic that _should_ be in the CASL ability but it isn't now, my plan is to refactor and centralise all the logic back in CASL.